crengine/src/lvpagesplitter.cpp

```diff
@@ -365,6 +365,10 @@ struct PageSplitState {
     }
     void SplitLineIfOverflowPage( LVRendLineInfo * line )
     {
+        // If page_h <= 0 this function will fall into an infinite loop
+        // in which it will allocate memory until it runs out.
+        if (page_h <= 0)
+            return;
```

Comment:
Do you remember the context/book with which this happened ?
Because it looks to me page_h is fixed and comes from ldomDocument._page_height , which is set by setRenderProps(... dy...) so very early in LVDocView from the screen/page height.
(May be it happened with table cells, floats, inline-box, and I forgot to pass this page_height from an upper context?)